include/mem_native.h

```diff
@@ -0,0 +1,115 @@
+/*
+ * SPDX-License-Identifier: GPL-2.0-or-later
+ *
+ * Copyright (C) 2020-2020  The dosbox-staging team
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ */
+
+#ifndef DOSBOX_MEM_NATIVE_H
+#define DOSBOX_MEM_NATIVE_H
+
+/*  These functions read and write 16, 32, and 64 bit uint's to
+ *  and from 8-bit memory positions using defined alignment. Use
+ *  these functions instead of C-style pointer type casts to larger
+ *  types, which are not alignment-safe.
+ */
+
+#include <cstdint>
+#include <cstring>
+
+/*  Read a uint16 from 8-bit native byte-ordered memory. Use this
+ *  instead of *(uint16_t*)(8_bit_ptr) or *(uint16_t*)(8_bit_ptr + offset)
+ */
+static inline uint16_t read_uint16(const uint8_t *arr)
+{
+	uint16_t val;
+	memcpy(&val, arr, sizeof(val));
+	return val;
+}
```

Comment:
Suggestion for naming:

`read_uint16` does not convey the meaning of why this function should be used instead pointer dereference - which might cause problem long-term for new people reading the code.

How about naming these functions with word `unaligned` to add a bit of context to the name, e.g.:
```C++
uint16_t x = read_unaligned_uint16(buf);
```
This would also make it easier to name the header file, it could be `mem_unaligned.h`.

Other options:
- `unaligned_read_uint16` (but I find `read_unaligned_*` to be more readable)
- `unaligned_readw` (has benefit of consistency with `host_read*` functions and is shorter)
- `read_unaligned<uint16>` (theoretically template would be most C++-like solution… but somehow I don't like it and would prefer to keep C-compatible interface)

Reply:
When with `read_unaligned_uint*`